dlt/extract/incremental/__init__.py

```diff
@@ -305,6 +315,16 @@ def _transform_item(
         row, start_out_of_range, end_out_of_range = transformer(row)
         self.start_out_of_range = start_out_of_range
         self.end_out_of_range = end_out_of_range
+        # if we know that rows are ordered we can close the generator automatically
+        # mind that closing pipe will not immediately close processing. it only closes the
+        # generator so this page will be fully processed
+        if self.row_order:
+            # ordered ascending, check if we cross upper bound
+            if self.row_order == "asc" and end_out_of_range:
+                self._pipe.close()
```

Comment:
There is an issue with parallelized, which is more of a problem with the paralellized wrapper.
Because of: https://github.com/dlt-hub/dlt/blob/rfix%2Fadds-row-order-incremental/dlt/extract/utils.py#L189

`incremental.bind()` is called with an empty pipe and this close call fails. I'm not sure how to fix it, any ideas?

Reply:
thanks for pointing out. we should not evaluate gens in the wrapper, only at runtime. I fixed `add_limit` to exactly do that... I need a test to reproduce it. there are other things that are passed in bind so it may be a bigger problem

Reply:
@steinitzu pls review commits below, they fix the late bind issue